mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -610,6 +610,26 @@ LogicalResult SYCLHostScheduleKernel::verify() {
   return verifyNdRange(*this, range, offset, ndRange);
 }
 
+void SYCLHostScheduleKernel::getEffects(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects) {
+  auto *defaultResource = SideEffects::DefaultResource::get();
+  // The nd-range arguments will be read from
+  if (Value range = getRange())
+    effects.emplace_back(MemoryEffects::Read::get(), range, defaultResource);
+  if (Value offset = getOffset())
+    effects.emplace_back(MemoryEffects::Read::get(), offset, defaultResource);
+  // The rest of the arguments will be scalar or have rw access
+  for (Value arg : getArgs()) {
+    if (!isa<LLVM::LLVMPointerType>(arg.getType())) {
+      // TODO: When we have accessor type information, drop write access for
+      // read_only accessors.
```

Comment:
It's a tricky one. I went for `Write|Read` because there is no way (AFAIK) to specify that we will be writing to the member pointer, so we'd be missing the `Write` semantics there, making MLIR passes infer wrong info and perform invalid optimizations, e.g., `canonicalize` and `cse` would mark the operation as dead and remove it, as it would have no memory effect and would yield no result.

Reply:
In this case, we should update/remove the `TODO` and add this reasoning in a comment.